st.c

```diff
@@ -4066,6 +4077,11 @@ kbdkey(void *data, struct wl_keyboard *keyboard, uint32_t serial, uint32_t time,
 	if (IS_SET(MODE_KBDLOCK))
 		return;
 
+	if (state == WL_KEYBOARD_KEY_STATE_PRESSED) {
+		wl.globalserial = serial;
+		return;
```

Comment:
Umm... you do realize that state is either pressed or released, and if you return early on both, the rest of the function won't run, right?

Anyway, I'll just squash your commits together and fix up some minor things so we don't have to continue this back and forth.